code/modules/mob/living/basic/space_fauna/headslug.dm

```diff
@@ -0,0 +1,97 @@
+/**
+ * ## Headslugs
+ *
+ * Player-controlled slugs that arise from a changeling ability in order to live on in an extremely limited capacity until they can find a suitable corpse to inhabit.
+ */
+/mob/living/basic/headslug
+	name = "headslug"
+	desc = "A small, slug-like creature with a large, gaping maw. It's covered in a thick, slimy mucus."
+	icon_state = "headslug"
+	icon_living = "headslug"
+	icon_dead = "headslug_dead"
+	gender = NEUTER
+	health = 50
+	maxHealth = 50
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomp"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+	faction = list(FACTION_CREATURE)
+	obj_damage = 0
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	speak_emote = list("squeaks")
+
+	ai_controller = /datum/ai_controller/basic_controller/headslug
+
+	/// The mind to transfer to our egg when it hatches
+	var/datum/mind/origin
+	/// Set to true once we've implanted our egg
+	var/egg_lain = FALSE
+
+/mob/living/basic/headslug/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	RegisterSignal(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET, PROC_REF(check_and_implant))
+
+/mob/living/basic/headslug/Destroy()
+	UnregisterSignal(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET)
+	return ..()
+
+/mob/living/basic/headslug/examine(mob/user)
+	. = ..()
+	if(isnull(client))
+		. += span_notice("It appears to be moving around listlessly.")
+	else
+		. += span_warning("It's moving around intelligently!")
```

Comment:
This is a weird examine, what's the reason behind it?

Reply:
because you can get empty changeling slugs (see ath's comment) and it's probably useful to know if it's empty or actually sapient so you don't squash it. it's just a newer thing that was introduced to compensate for some stuff that was added during the simple_animal port, lmk if i should remove it